pkg/api/datasources.go

```diff
@@ -577,36 +565,9 @@ func (hs *HTTPServer) UpdateDataSourceByUID(c *contextmodel.ReqContext) response
 	}
 	cmd.ID = ds.ID
 
-	// check if LBAC rules have been modified
-	hasAccess, errAccess := checkTeamHTTPHeaderPermissions(hs, c, ds, cmd)
-	if !hasAccess {
-		return response.Error(http.StatusForbidden, fmt.Sprintf("You'll need additional permissions to perform this action. Permissions needed: %s", datasources.ActionPermissionsWrite), errAccess)
-	}
-
 	return hs.updateDataSourceByID(c, ds, cmd)
 }
 
-func getEncodedString(jsonData *simplejson.Json, key string) string {
-	if jsonData == nil {
-		return ""
-	}
-	jsonValues, exists := jsonData.CheckGet(key)
-	if !exists {
-		return ""
-	}
-	val, _ := jsonValues.Encode()
-	return string(val)
-}
-
-func checkTeamHTTPHeaderPermissions(hs *HTTPServer, c *contextmodel.ReqContext, ds *datasources.DataSource, cmd datasources.UpdateDataSourceCommand) (bool, error) {
```

Comment:
we can remove this because we are overriding rules, incase they are present. No longer needed to check permissions, since we are not using these endpoints for team lbac.